configuration/configuration.go

```diff
@@ -202,6 +202,9 @@ type Configuration struct {
 				Deny []string `yaml:"deny,omitempty"`
 			} `yaml:"urls,omitempty"`
 		} `yaml:"manifests,omitempty"`
+		// RepositoryClass validation checks that the class of the configurations inside the
+		// the manifest matches the authorized resource when provided.
+		RepositoryClass bool `yaml:"repositoryclass,omitempty"`
```

Comment:
We could create a policy section for options like these. However I don't see it as too much different from the URLs allow/deny, also policies.